src/InsertionsClient/Api/Props/PropsVariableDeducer.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Net.Insertions.Api.Models;
+using Microsoft.Net.Insertions.Models;
+using Microsoft.Net.Insertions.Props.Models;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Packaging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.Net.Insertions.Api
+{
+    /// <summary>
+    /// Deduces the values of given props file variables by matching:
+    /// <list type="bullet">
+    /// <item> Paths to files extracted from nuget packages into the folders specified in default.config and .packageconfig files </item>
+    /// <item> Paths to files specified in swr files. </item>
+    /// </list>
+    /// </summary>
+    internal class PropsVariableDeducer
```

Comment:
design question: feels like this class should be in a distinct namespace, as it's essentially a client to the packaging web api.  (very nice, by the way)

This may be something we think about after this PR merge.